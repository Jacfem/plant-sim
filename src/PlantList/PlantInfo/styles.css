.unstyledList {
  list-style: none;
  padding-left: 0; /* unset hidden padding from list style */
}

.showOnHover { /* very quick 'hide and show' effect to make seedData less obvious */
  background-color: light-gray;
  opacity: 0;
}

.showOnHover:hover {
  opacity: 1;
}
