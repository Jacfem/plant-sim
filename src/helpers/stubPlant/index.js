export const stubPlant = {
  plant: {
    key: '0',
    name: 'Stubbed Plant',
    needs: {
      fertilizer: 8,
      sunlight: 5,
      water: 9,
    },
    picture: 'picture.url',
    slug: 'stubbed-plant',
  },
};
