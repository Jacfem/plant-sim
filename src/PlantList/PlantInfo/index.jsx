import React from 'react';
import { Link } from 'react-router-dom';

import './styles.css';

const PlantInfo = props => (
  <li key={props.plant.key}>
    <div className="plantContainer">
      <h3>{props.plant.name}:
        <Link to={`/plant/${props.plant.slug}`}>
          Try it!
        </Link>
      </h3>
      <p>Water:
        <span className="showOnHover">
          {props.plant.needs.water}
        </span>
      </p>
      <p>Days needed:
        <span className="showOnHover">
          {props.plant.needs.sunlight}
        </span>
      </p>
      <p>Fertilizer:
        <span className="showOnHover">
          {props.plant.needs.fertilizer}
        </span>
      </p>
    </div>
  </li>
);

// Add propTypes required here

export default PlantInfo;
