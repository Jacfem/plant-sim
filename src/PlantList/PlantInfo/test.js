import PlantInfo from './';
import { stubPlant } from '../../helpers/stubPlant';

describe('<PlantInfo />', () => {
  function subject() {
    const props = {
      ...stubPlant,
    };
    return Enzyme.shallow(<PlantInfo {...props} />);
  }

  describe('#render', () => {
    it('matches snapshot', () => {
      expect(subject()).toMatchSnapshot();
    });
  });
});
