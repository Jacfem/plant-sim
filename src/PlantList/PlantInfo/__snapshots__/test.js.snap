// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlantInfo /> #render matches snapshot 1`] = `
<li
  key="0"
>
  <div
    className="plantContainer"
  >
    <h3>
      Stubbed Plant
      :
      <Link
        replace={false}
        to="/plant/stubbed-plant"
      >
        Try it!
      </Link>
    </h3>
    <p>
      Water:
      <span
        className="showOnHover"
      >
        9
      </span>
    </p>
    <p>
      Days needed:
      <span
        className="showOnHover"
      >
        5
      </span>
    </p>
    <p>
      Fertilizer:
      <span
        className="showOnHover"
      >
        8
      </span>
    </p>
  </div>
</li>
`;
